src/AutoRest.CSharp.V3/Output/Builders/ClientBuilder.cs

```diff
@@ -429,16 +424,16 @@ private static RestClientMethod BuildNextPageMethod(RestClientMethod method)
                 "nextLink",
                 "The URL to the next page of results.",
                 typeof(string),
-                null,
-                true);
-            List<Parameter> parameters = new List<Parameter>();
-            parameters.Add(nextPageUrlParameter);
+                defaultValue: null,
+                isRequired: true,
+                isArtificial: true);
+            List<Parameter> parameters = new List<Parameter> { nextPageUrlParameter };
             parameters.AddRange(method.Parameters.Where(p => p.Name != nextPageUrlParameter.Name));
 
             var request = new Request(
                 method.Request.HttpMethod,
-                new[] { new PathSegment(nextPageUrlParameter, false, SerializationFormat.Default),  },
-                Array.Empty<PathSegment>(),
+                method.Request.HostSegments,
+                new[] { new PathSegment(nextPageUrlParameter, false, SerializationFormat.Default) },
```

Comment:
We need host segments now, and the artificial nextLink parameter becomes a path parameter.

Reply:
Let's merge host and paths segments together separation is not useful anymore.

Reply:
So, have Segment specify if it is host or path?

Reply:
Do we even care about the distiction?

Reply:
Well, host segments use WriteUriFragment and path segments use WritePathSegment, and those use AppendRaw and AppendPath respectively.

Reply:
If I make the bigger changes to AppendRaw to also do path and query params too, then it might not need to use 2 different methods? I'd have to test it.

Reply:
We should use non-Raw methods where possible. Might need to add some bool to the segment

Reply:
Done. Combined collection of host and path segments together. They are only different if the segment has `IsRaw` as true.